haystack/agents/base.py

```diff
@@ -156,8 +174,32 @@ def __init__(
         self.max_steps = max_steps
         self.tool_pattern = tool_pattern
         self.final_answer_pattern = final_answer_pattern
+
+        self.add_default_logging_callbacks()
+
         send_custom_event(event=f"{type(self).__name__} initialized")
 
+    def add_default_logging_callbacks(self, agent_color: Color = Color.GREEN) -> None:
+        def on_tool_finish(
+            tool_output: str,
+            color: Optional[Color] = None,
+            observation_prefix: Optional[str] = None,
+            llm_prefix: Optional[str] = None,
+            **kwargs: Any,
+        ) -> None:
+            print_text(observation_prefix)  # type: ignore
+            print_text(tool_output, color=color)
+            print_text(f"\n{llm_prefix}")
+
+        def on_agent_start(**kwargs: Any) -> None:
+            agent_info_copy = kwargs.copy()
```

Comment:
I don't think you need to copy, it's ok to alter `kwargs`, see https://docs.python.org/3/reference/compound_stmts.html#function-definitions